packages/auth/src/next/routes/logout.ts

```diff
@@ -1,45 +1,51 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+
 import { getActiveCookie, getUser } from "../helpers/user";
 import { ThirdwebAuthContext } from "../types";
-import { serialize } from "cookie";
-import { NextApiRequest, NextApiResponse } from "next";
 
 export default async function handler(
-  req: NextApiRequest,
-  res: NextApiResponse,
+  req: NextRequest,
   ctx: ThirdwebAuthContext,
 ) {
   if (req.method !== "POST") {
-    return res.status(405).json({
-      error: "Invalid method. Only POST supported.",
-    });
+    return Response.json(
+      { error: "Invalid method. Only POST supported." },
+      { status: 405 },
+    );
   }
 
-  const activeCookie = getActiveCookie(req);
+  const activeCookie = getActiveCookie();
   if (!activeCookie) {
-    return res.status(400).json({
-      error: "No logged in user to logout.",
-    });
+    return Response.json(
+      { error: "No logged in user to logout." },
+      { status: 400 },
+    );
   }
 
   if (ctx.callbacks?.onLogout) {
-    const user = await getUser(req, ctx);
+    const user = await getUser(ctx, req);
     if (user) {
       await ctx.callbacks.onLogout(user, req);
     }
   }
 
   // Set the access token to 'none' and expire in 5 seconds
-  res.setHeader(
-    "Set-Cookie",
-    serialize(activeCookie, "", {
-      domain: ctx.cookieOptions?.domain,
-      path: ctx.cookieOptions?.path || "/",
-      sameSite: ctx.cookieOptions?.sameSite || "none",
-      expires: new Date(Date.now() + 5 * 1000),
-      httpOnly: true,
-      secure: ctx.cookieOptions?.secure || true,
-    }),
+  const response = NextResponse.json(
+    { message: "Successfully logged out." },
+    { status: 200 },
   );
 
-  return res.status(200).json({ message: "Successfully logged out" });
+  response.cookies.set({
+    name: activeCookie,
+    value: '',
+    domain: ctx.cookieOptions?.domain,
+    path: ctx.cookieOptions?.path || "/",
+    sameSite: ctx.cookieOptions?.sameSite || "none",
+    expires: new Date(Date.now() + 5 * 1000),
```

Comment:
@adam-maj I wonder why setting the cookie expiry to 5 seconds instead of deleting the cookie at that time?

Reply:
@adam-maj ping